text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <group_beam_searcher.hpp>
+#include <openvino/openvino.hpp>
+#include <openvino_extensions/strings.hpp>
+
+namespace {
+std::pair<ov::Tensor, ov::Tensor> tokenize(ov::InferRequest&& tokenizer, std::string_view prompt) {
+    ov::Tensor destination = tokenizer.get_input_tensor();
+    openvino_extensions::pack_strings(std::array{prompt}, destination);
+    tokenizer.infer();
+    return {tokenizer.get_tensor("input_ids"), tokenizer.get_tensor("attention_mask")};
+}
+
+std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>& tokens) {
+    constexpr size_t BATCH_SIZE = 1;
+    ov::Tensor inp = detokenizer.get_input_tensor();
+    inp.set_shape({BATCH_SIZE, tokens.size()});
+    for (size_t idx = 0; idx < tokens.size(); ++idx) {
+        inp.data<int64_t>()[idx] = tokens.at(idx);
+    }
+    detokenizer.infer();
+    return openvino_extensions::unpack_strings(detokenizer.get_output_tensor()).front();
+}
+}
+
+int main(int argc, char* argv[]) try {
+    if (argc != 5) {
+        throw std::runtime_error(std::string{"Usage: "} + argv[0]
+            + " <openvino_model.xml> <tokenizer.xml> <detokenizer.xml> '<prompt>'");
+    }
+    ov::Core core;
+    core.add_extension(USER_OV_EXTENSIONS_PATH);  // USER_OV_EXTENSIONS_PATH is defined in root CMakeLists.txt
+    auto [input_ids, mask] = tokenize(core.compile_model(argv[2], "CPU").create_infer_request(), argv[4]);
```

Comment:
It is nice to combine everything in a single statement because it needs calling tokenizer only once _in this particular sample_. But let's make it just more practical for code reuse. In real application, most likely, tokenizer will be used multiple times, and if the user doesn't really understand what's happening here in this statement, this line -- without much modifications -- will be used multiple times with the compile_model, creation of infer request and disposal, that is not really what we want to teach users.

Reply:
Done